src/main/java/picard/sam/AbstractAlignmentMerger.java

```diff
@@ -966,6 +1029,12 @@ public boolean isClipOverlappingReads() {
 
     public void setClipOverlappingReads(final boolean clipOverlappingReads) {
         this.clipOverlappingReads = clipOverlappingReads;
+        this.hardClipOverlappingReads = false;
```

Comment:
I know I said this is better than changing a public API...but it's still not good. `set*` methods should only change the variable they are named for. why not add a `setClipWithHardClips` method and avoid changing two things in one method?